modules/hub/src/OnchainTransactionService.ts

```diff
@@ -333,6 +332,53 @@ export class OnchainTransactionService {
       return
     }
 
+    if (txn.state == 'pending_failure') {
+      const [tx, err] = await maybe(this.web3.eth.getTransaction(txn.hash))
+      LOG.info('State of {txn.hash}: {res}', {
+        txn,
+        res: JSON.stringify(tx || err),
+      })
+      if (err) {
+        // TODO: what errors can happen here?
+        LOG.warning(`Error checking status of tx '${txn.hash}': ${'' + err} (will retry)`)
+        return
+      }
+
+      if (tx) {
+        // if tx exists at this point, mark as submitted
+        await this.updateTxState(txn, {
+          state: 'submitted',
+        })
```

Comment:
There should be a return after this, since we should never check timeouts etc if we've seen the tx on chain.